app/javascript/components/student/Editor.tsx

```diff
@@ -11,12 +11,13 @@ import { FileEditor, FileEditorHandle } from './editor/FileEditor'
 import { fetchJSON } from '../../utils/fetch-json'
 import { typecheck } from '../../utils/typecheck'
 import { Iteration } from '../track/IterationSummary'
+import { useIsMounted } from 'use-is-mounted'
 
 export type Submission = {
   testsStatus: SubmissionTestsStatus
   uuid: string
   links: SubmissionLinks
-  testRun: TestRun
+  testRun?: TestRun
```

Comment:
FYI, this does *not* means `testRun` can be `TestRun` or `undefined`. It means: `testRun` may not be present. Doesn't matter here, but might at some point in the future.